### PR TITLE
Add default primary column name to list table classes. #3492.

### PR DESCRIPTION
**Pull request by mindctrl**
#3492

### CODE REVIEW

File: includes/admin/payments/class-payments-table.php [hunk 1]
```diff
@@ -247,6 +247,18 @@ public function get_sortable_columns() {
...
 	/**
+	 * Gets the name of the default primary column.
+	 *
+	 * @since 2.5
+	 * @access protected
+	 *
+	 * @return string Name of the default primary column.
+	 */
+	protected function get_default_primary_column_name() {
+		return 'ID';
```
Comment (pippinsplugins) [inline includes/admin/payments/class-payments-table.php]: I think this should be the email column. ID isn't really important

File: includes/admin/reporting/class-sales-logs-list-table.php [hunk 2]
```diff
@@ -52,6 +52,18 @@ public function __construct() {
...
 	/**
+	 * Gets the name of the default primary column.
+	 *
+	 * @since 2.5
+	 * @access protected
+	 *
+	 * @return string Name of the default primary column.
+	 */
+	protected function get_default_primary_column_name() {
+		return 'ID';
```
Comment (pippinsplugins) [inline includes/admin/reporting/class-sales-logs-list-table.php]: This should be the download name I think.

Comment (mindctrl) [inline includes/admin/reporting/class-sales-logs-list-table.php]: @pippinsplugins I agree. I originally tried that but it looks like this:

image

Same for some of the other ones. Thoughts about that? Seems like any 'primary' column that isn't the first one shows up weird like that. Am I missing something?

Comment (mindctrl) [inline includes/admin/reporting/class-sales-logs-list-table.php]: ^ my comment was meant for the payments table file. But same thing applies here.

Comment (pippinsplugins) [inline includes/admin/reporting/class-sales-logs-list-table.php]: That looks like it has two primary columns. When collapsed it should only have one column. That's weird.

File: includes/admin/payments/class-payments-table.php (hunk 1 above)
Comment (cklosowski) [inline includes/admin/payments/class-payments-table.php]: I think I disagree here. Payments are made in chronological order, and setting the email as the primary column doesn't achieve us much, as it's not something you'd really sort on either. The ID is the unique value in the payments world, not the email.

**Comment by cklosowski** [comment]
Shouldn't we be using `get_primary_column_name` not `get_default_primary_column_name`? Must like sortable columns, the items in the extended class shouldn't use the `default` getter methods.

**Comment by cklosowski** [comment]
Just realized this was to be into 2.5, going to close this and repoen it for 2.5

**Pull request closed without merging**